apps/explorer/tests/test_views.py

```diff
@@ -50,3 +50,230 @@ def test_renders_pixelset_list(self):
             '<tr class="pixelset">',
             count=10
         )
+
+    def test_species_filter(self):
+
+        # Create 8 pixelset
+        make_development_fixtures(
+            n_pixel_sets=8,
+            n_pixels_per_set=1
+        )
+
+        # Create our special ones
+        species = factories.SpeciesFactory()
+        strain = factories.StrainFactory(species=species)
+        n_new_pixel_sets = 2
+        pixel_sets = factories.PixelSetFactory.create_batch(n_new_pixel_sets)
+
+        omics_units = factories.OmicsUnitFactory.create_batch(
+            n_new_pixel_sets,
+            strain=strain
+        )
+        for omics_unit, pixel_set in zip(omics_units, pixel_sets):
+            factories.PixelFactory(
+                omics_unit=omics_unit,
+                pixel_set=pixel_set,
+            )
+
+        # no filter
+        response = self.client.get(self.url)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=10
+        )
+
+        # species filter
+        data = {
+            'species': [species.id, ]
+        }
+        response = self.client.get(self.url, data)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=n_new_pixel_sets
+        )
+
+    def test_omics_unit_types_filter(self):
+
+        # We generate one pixel per pixelset so that half of pixel sets
+        # contains pixels with 'mRNA' omics unit type and the other half
+        # 'protein' omics unit type
+        n_pixel_sets = 8
+        make_development_fixtures(
+            n_pixel_sets=n_pixel_sets,
+            n_pixels_per_set=1
+        )
+
+        # no filter
+        response = self.client.get(self.url)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=n_pixel_sets
+        )
+
+        # omics_unit_types filter
+        omics_unit_type = models.OmicsUnitType.objects.first()
+        data = {
+            'omics_unit_types': [omics_unit_type.id, ]
+        }
+        response = self.client.get(self.url, data)
+        self.assertContains(
+            response,
+            '<tr class="pixelset">',
+            count=int(n_pixel_sets / 2)
```

Comment:
Isnt it a `int` by default in Python 3?

Reply:
Yes my lord. Old habits.

Reply:
Meh… answered too quickly. In fact that's the opposite: this is the case in python 2, python 3 always returns a `float` as a division result.

Reply:
ah, then is it... `//` that we should be using now?